py3status/modules/gazelle.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""
+Show your ratio on your favorite Gazelle-based private tracker.
+
+This module can get various useful infos about you on a private tracker.
+You can display your ratio, upload size, download size.
+The only requirements are that the private tracker is based on the Gazelle framework,
+for instance Apollo or Redacted, and that the sysops didn't block API access somehow.
+This module will save session cookies to avoid too much opened sessions which could
+potentially put you into troubles due to a suspect activity.
+
+Configuration parameters:
+    baseurl: Base URL of the tracker (example: https://what.cd) (default None)
+    cache_timeout: How often we refresh this module in seconds (default 60)
+    format: See placeholders below (default 'Ratio: {ratio}')
+    password: Account password (default '')
+    thresholds: specify color thresholds to use
+        (default [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')])
+    urgent_new_messages: set output urgent mode when there are new messages
+        (default False)
+    urgent_new_notifs: set output urgent mode when there are new notifications
+        (default False)
+    username: Account username (default '')
+
+Format placeholders:
+    {ratio} account ratio
+    {download} account download amount
+    {upload} account upload amount
+    {messages} number of unread messages
+    {notifs} number of new notifications
+
+@author raspbeguy <raspbeguy@hashtagueule.fr>
+@license MIT
+"""
+
+from py3status.exceptions import RequestException
+
+import re
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    baseurl = None
+    cache_timeout = 60
+    format = 'Ratio: {ratio}'
+    password = ''
+    thresholds = [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')]
+    urgent_new_messages = False
+    urgent_new_notifs = False
+    username = ''
+
+    def _readable_size(self, size):
+        num = size
+        for unit in ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB']:
+            if abs(num) < 1024.0:
+                return "%3.1f %s" % (num, unit)
```

Comment:
We could use `self.py3.format_units` instead. Expose units as placeholders too for more flexible customization, eg `[\?if=max_length 1 {download_unit}]` to print `K`, `M`, `G`, etc. Also, the cutoff is at 1000. File a bug / PR if you want to change that. File manager `nautilus` (K->M) cutoff seems to be at 977 for me... saying it's `1.0 MB`.

Reply:
I really need the 1024 base and I'm not willing to wait for another PR to be accepted. Anyway, this can be done later.
I think that having a placeholder for units is overkill.

Reply:
It is all fun and games until someone can't customize that one damn thing.

P.S. Your placeholders also messes with the rounding. We can't do this: `{download:d}`, `{download:.0f}`, `{upload:.1f}`, or *New!* `{download:g}`.

Reply:
So what? the placeholder bullshit syntax makes already the `format` parameter unreadable, I'm not willing to make it even less readable by adding some useless placeholders.